### ISSUE TITLE
Feature Request: ETW Hooks for Windows

### ISSUE THREAD

**Issue by sempervictus** [wazuh/wazuh#16668]
[Event Tracing for Windows ](https://www.ired.team/miscellaneous-reversing-forensics/windows-kernel-internals/etw-event-tracing-for-windows-101) can be used to observe and report on granular details of system behaviors. It may be beneficial for Windows-resident agents to be able to configure, start, and collect traces (with appropriate cleanup) so as to rapidly adapt to changing IOCs through rules/agent config updates.

**Comment by Dwordcito** [comment]
Hi @sempervictus 
For some time now we have been analyzing to incorporate ETW as an event provider for Windows.

Basically, we need it for many things (one for example is that syscollector works in real-time and not in a scheduled way as it does now).

Here's a spike we made, which identifies the feasibility of incorporating it through the cross-compilation we used (unfortunately krabsetw is not a viable option.)

On behalf of my team, we will prioritize this task when we finish the vulnerability detector refactor.

https://github.com/wazuh/wazuh/issues/11723
